java_fundamentals_block/optional_task/src/main/java/OptionalTask1.java

```diff
@@ -0,0 +1,82 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class OptionalTask1 {
+
+    private static BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+
+    public static int[] inputElem() throws IOException {
+        System.out.println("Enter the number of elements:");
+        int numberOfElements = Integer.parseInt(reader.readLine());
+        int[] array = new int[numberOfElements];
+        System.out.println("Enter the elements:");
+        for (int i = 0; i < numberOfElements; i++) {
+            array[i] = Integer.parseInt(reader.readLine());
+        }
+        return array;
+    }
+
+    public static void findShortElement(int[] array) {
```

Comment:
I think in current implementation these functions do 2 things:
1. action (like find shortest element here)
2. printing to console

I suggest leave only single responsibility to the function and move printing out to `OptionalTaskMain.java`

so converted this function will be like
```    
public static int findShortestElement(int[] array) {
        int shortElem = array[0];
        for (int i = 1; i < array.length; i++) {
            if (array[i] < shortElem) {
                shortElem = array[i];
            }
        }
   return shortElem;
}
```

And then in OptionalTaskMain
```
int shortestElem = OptionalTask1.findShortElement(array);
        System.out.println("The longest element is: " + shortestElem);
        System.out.println("The length of this element is: " + OptionalTask1.elemLength(shortestElem));
```

Reply:
Done for this and other methods